python/dgl/data/yelp.py

```diff
@@ -0,0 +1,164 @@
+"""Yelp Dataset"""
+import os
+import json
+import numpy as np
+import scipy.sparse as sp
+from .. import backend as F
+from ..convert import from_scipy
+from ..transforms import reorder_graph
+from .dgl_dataset import DGLBuiltinDataset
+from .utils import generate_mask_tensor, load_graphs, save_graphs, _get_dgl_url, download, \
+    extract_archive
+
+
+class YelpDataset(DGLBuiltinDataset):
+    r"""Yelp dataset for node classification from `GraphSAINT: Graph Sampling Based Inductive
+        Learning Method <https://arxiv.org/abs/1907.04931>`_
+
+        The task of this dataset is categorizing types of businesses based on customer reviewers and
+        friendship.
+
+        Yelp dataset statistics:
+
+        - Nodes: 716,847
+        - Edges: 13,954,819
+        - Number of classes: 100 (Multi-class)
+        - Node feature size: 300
+
+        Parameters
+        ----------
+        raw_dir : str
+            Raw file directory to download/contains the input data directory.
+            Default: ~/.dgl/
+        force_reload : bool
+            Whether to reload the dataset.
+            Default: False
+        verbose : bool
+            Whether to print out progress information.
+            Default: False
+        transform : callable, optional
+            A transform that takes in a :class:`~dgl.DGLGraph` object and returns
+            a transformed version. The :class:`~dgl.DGLGraph` object will be
+            transformed before every access.
+        reorder : bool
+            Whether to reorder the graph using :func:`~dgl.reorder_graph`.
+            Default: False.
+
+        Attributes
+        ----------
+        num_classes : int
+            Number of node classes
+
+        Examples
+        --------
+        >>> dataset = YelpDataset()
+        >>> dataset.num_classes
+        100
+        >>> g = dataset[0]
+        >>> # get node feature
+        >>> feat = g.ndata['feat']
+        >>> # get node labels
+        >>> labels = g.ndata['label']
+        >>> # get data split
+        >>> train_mask = g.ndata['train_mask']
+        >>> val_mask = g.ndata['val_mask']
+        >>> test_mask = g.ndata['test_mask']
```

Comment:
Have you tried this example locally?

Reply:
Yes, I have tried the example locally using dgl build from source with this PR.